src/Layout.js

```diff
@@ -68,14 +68,8 @@ var computeLayout = (function() {
       var location = locations[i];
 
       var key = type + capitalizeFirst(location) + suffix;
-      if (key in node.style) {
-        return node.style[key];
-      }
-
-      key = type + suffix;
-      if (key in node.style) {
-        return node.style[key];
-      }
+  
+        return node.style[key] || node.style[type + key + suffix];
```

Comment:
A safe way to do it is by writing `node.style['marginTop'] !== undefined`. But, in order to make sure that it has performance improvement, we need to benchmark it.

Can you use the code that generates a random test to generate a big random tree and then time the layout algorithm before and after? Otherwise we can't know if this is a performance optimization or not.